yal/main.py

```diff
@@ -147,6 +147,20 @@ async def process_message(self, message):
                 message.transport_metadata, "message", "button", "payload"
             )
 
+            whatsapp_delivery_failed = self.user.metadata.get(
```

Comment:
Can we move some of this into a separate `reset_whatsapp_delivery_failure` function